crates/primitives/src/transaction/mod.rs

```diff
@@ -620,6 +620,75 @@ impl TransactionSigned {
         initial_tx.hash = initial_tx.recalculate_hash();
         initial_tx
     }
+
+    /// Decodes a eip2718 transaction where tx is [id, rlp(typedtx)]
+    pub fn decode_enveloped(&self, tx: Bytes) -> Result<Self, DecodeError> {
+        let mut data = tx.as_ref();
+
+        let first_header = Header::decode(&mut data)?;
+
+        if data[0] > 0x7f {
```

Comment:
I would use the rlp constant here instead:
https://github.com/paradigmxyz/reth/blob/780419fb0b0b6e9327d706c9136365a9a776b92b/crates/rlp/src/types.rs#L7